torch/utils/data/dataloader.py

```diff
@@ -479,6 +479,9 @@ def __len__(self) -> int:
             # Cannot statically verify that dataset is Sized
             length = self._IterableDataset_len_called = len(self.dataset)  # type: ignore[assignment, arg-type]
             if self.batch_size is not None:  # IterableDataset doesn't allow custom sampler or batch_sampler
+                if self.batch_size > length:
```

Comment:
Is it necessary to enforce users to make `batch_size` smaller than `length`? With `drop_last` as `False`, users can still access a non-filled batch.

Reply:
I think its necessary because if its smaller than lenght, lenght will be equel to zero. And dataloader return no value in for loop.

Reply:
> I think its necessary because if its smaller than lenght, lenght will be equel to zero.

When `drop_last` is `False`, the length is 1. I know it's a little weird to have a single batch from `DataLoader` but it's still legit to do so.

Reply:
Okey in that case we need this check when drop_last is True?

Reply:
Sure